### DISCUSSION TITLE
What can be stored in pass file?

### DISCUSSION THREAD

**Discussion by doolio** [languitar/pass-git-helper#322] [Q&A] [answered]
I want to use this git credential helper for `git send-email` rather than have `smtpPass` in my `git` config file in clear text. Is it possible to store the other `smtp` settings in the pass file as well. Such as:

```pass
<smtpPass>
smtpencryption: tls         
smtpserver: smtp.gmail.com   
smtpserverport: 587          
smtpuser: <username>@gmail.com  
```

That is am I right in assuming that `pass-git-helper $@` will pick up all these parameters? Thanks.

**Comment by doolio** [comment]
I think I can add the following to my `git-pass-mapping.ini` and thus include `smtpuser` in my `password-store` file Not sure about the other smtp settings.

```ini
[smtp.gmail.com*]
username_extractor=regex_search
regex_username=^smtpuser: +(.*)$
```

**Reply by languitar** [reply, marked as answer]
pass-git-helper is specifically tailored to fulfill the [git credential protocol](https://git-scm.com/docs/git-credential). That protocol doesn't know about smtp settings.

**Reply by doolio** [reply]
Ah I see. I was lead to believe by another discussion online that this helper was what I needed. Anyway, thanks for your time and the prompt reply.
